okhttp/src/nonJvmMain/kotlin/okhttp3/RequestBody.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2014 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3
+
+import okhttp3.internal.commonContentLength
+import okhttp3.internal.commonIsDuplex
+import okhttp3.internal.commonIsOneShot
+import okhttp3.internal.commonToRequestBody
+import okio.BufferedSink
+import okio.ByteString
+import okio.internal.commonAsUtf8ToByteArray
+
+actual abstract class RequestBody {
+  actual abstract fun contentType(): MediaType?
+
+  actual open fun contentLength(): Long = commonContentLength()
+
+  actual abstract fun writeTo(sink: BufferedSink)
+
+  actual open fun isDuplex(): Boolean = commonIsDuplex()
+
+  actual open fun isOneShot(): Boolean = commonIsOneShot()
+
+  actual companion object {
+    actual fun String.toRequestBody(contentType: MediaType?): RequestBody {
+      val bytes = commonAsUtf8ToByteArray()
+      // TODO Consider adding charset UTF-8 to contentType
```

Comment:
We should do this before this PR lands. Otherwise the two platforms are inconsistent?